File: testData/src/java8/pkg/TestTrySplit.java
package pkg;

public class TestTrySplit {
  public void test() {
    Object obj = null;
    try {
      obj = new Object();
    } catch (ArithmeticException ex) {
      if (obj != null) {
        System.out.println("a");
      }

      throwMyException(ex.getMessage());
      return;
    } finally {
      System.out.println("b");
    }
  }

  public void testFlat() {
    Object obj = null;
    b1:
    {
      b2:
      try {
        try {
          obj = new Object();
        } catch (ArithmeticException ex) {
          if (obj != null) {
            System.out.println("a");
          }

          throwMyException(ex.getMessage());
          break b2;
        }
        break b1;
      } catch (Throwable t) {
        System.out.println("b");
        throw t;
      }
      System.out.println("b");
      return;
    }
    System.out.println("b");
  }

  public static void throwMyException(String message) {
    throw new RuntimeException(message);
  }
}

File: plugins/kotlin/testData/src/kt/pkg/TestTrySplit.kt
package pkg

class TestTrySplit {
  fun test() {
    var obj: Any? = null
    try {
      obj = Any()
    } catch (ex: ArithmeticException) {
      if (obj != null) {
        println("a")
      }

      throwMyException(ex.message)
      return
    } finally {
      println("b")
    }
  }

  fun testFlat() {
    var obj: Any? = null
    run b1@{
      run b2@{
        try {
          try {
            obj = Any()
          } catch (ex: ArithmeticException) {
            if (obj != null) {
              println("a")
            }

            throwMyException(ex.message)
            return@b2
          }
          return@b1
        } catch (t: Throwable) {
          println("b")
          throw t
        }
      }
      println("b")
      return
    }


    println("b")
  }

  companion object {
    fun throwMyException(message: String?) {
      throw RuntimeException(message)
    }
  }
}
